library/core/src/pin.rs

```diff
@@ -733,7 +733,7 @@
 //! sensitivity, it will need to be structurally pinned.
 //!
 //! A useful test is if [`unsafe`] code that consumes <code>[Pin]\<[&mut Struct][&mut]></code>
-//! also needs to take note of the address of the field itself, it may be evidence that that field
+//! also needs to take note of the address of the field itself, it may be evidence that field
```

Comment:
Problem: `that` is not redundant here, because it is trying to be precise in which field it is talking about.

Reply:
If you're curious, you can find a lot of discussion of this construct online. For example: https://english.stackexchange.com/questions/3418/how-do-you-handle-that-that-the-double-that-problem

Personally I avoid "that that" _because_ it looks like a mistake. But a quick grep shows this is clearly not common in the compiler.

Reply:
“that this” is sometimes a suitable replacement for “that that” that is specific but not redundant, I think it could work here (but haven’t read the full sentence)